drivers/SmartThings/zwave-switch/src/multi-metering-switch/init.lua

```diff
@@ -0,0 +1,117 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local st_device = require "st.device"
+local capabilities = require "st.capabilities"
+--- @type st.zwave.CommandClass
+local cc = require "st.zwave.CommandClass"
+--- @type st.zwave.CommandClass.Meter
+local Meter = (require "st.zwave.CommandClass.Meter")({version = 3})
+--- @type st.zwave.CommandClass.Basic
+local Basic = (require "st.zwave.CommandClass.Basic")({ version = 1, strict = true })
+--- @type st.zwave.CommandClass.SwitchBinary
+local SwitchBinary = (require "st.zwave.CommandClass.SwitchBinary")({version = 2, strict = true })
+local MULTI_METERING_SWITCH_CONFIGURATION_MAP = require "multi-metering-switch/multi_metering_switch_configurations"
+
+local PARENT_ENDPOINT = 1
+
+local MULTI_METERING_SWITCH_FINGERPRINTS = {
+  {mfr = 0x0086, prod = 0x0003, model = 0x0084}, -- Aeotec Nano Switch 1
+  {mfr = 0x0086, prod = 0x0103, model = 0x0084}, -- Aeotec Nano Switch 1
+  {mfr = 0x0086, prod = 0x0203, model = 0x0084}, -- AU Aeotec Nano Switch 1
+  {mfr = 0x027A, prod = 0xA000, model = 0xA004}, -- Zooz ZEN Power Strip 1
+  {mfr = 0x015F, prod = 0x3102, model = 0x0201}, -- WYFY Touch 1-button Switch
+  {mfr = 0x015F, prod = 0x3102, model = 0x0202}, -- WYFY Touch 2-button Switch
+  {mfr = 0x015F, prod = 0x3102, model = 0x0204}, -- WYFY Touch 4-button Switch
+  {mfr = 0x015F, prod = 0x3111, model = 0x5102}, -- WYFY Touch 1-button Switch
+  {mfr = 0x015F, prod = 0x3121, model = 0x5102}, -- WYFY Touch 2-button Switch
+  {mfr = 0x015F, prod = 0x3141, model = 0x5102}, -- WYFY Touch 4-button Switch
+}
+
+local function can_handle_multi_metering_switch(opts, driver, device, ...)
+  for _, fingerprint in ipairs(MULTI_METERING_SWITCH_FINGERPRINTS) do
+    if device:id_match(fingerprint.mfr, fingerprint.prod, fingerprint.model) then
+      return true
+    end
+  end
+  return false
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    local children_amount = MULTI_METERING_SWITCH_CONFIGURATION_MAP.get_child_amount(device)
+    for i = 2, children_amount+1, 1 do
```

Comment:
The zwave-multi-metering-switch [DTH](https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/78747fa5a3b42a0c801f1095f8f7521da968ba42/devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy#L374) creates child switches for values [2, numberOfSwitches]. In a case where number of switches = 5, this would then create switches on endpoint 2, 3, 4, 5 (4 total switches).

In this LUA driver, it creates child switches for values [2, childrenAmount + 1] and therefore results in 5 child switches. Is this change in behavior expected?

Reply:
If this behavior is expected, then would we need to decrease the number of children? For example in the DTH the Zooz Zen Strip has 4 child switches created (as explained above): https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/78747fa5a3b42a0c801f1095f8f7521da968ba42/devicetypes/smartthings/zwave-multi-metering-switch.src/zwave-multi-metering-switch.groovy#L107

Here in this driver, the amount of children for the Zooz Zen Strip is set to 5, which results in 5 switches, so should the number of children be decreased?

Reply:
Yes, you're correct. 
I decreased the number of children in configuration file, so now the driver creates child switches according to DTH with correct amount.